File: apps/playmaker/generate_splash_logos.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

// Simple PNG generator using Canvas (will check if available)
async function generateLogos() {
    console.log('🎨 Generating splash screen logos...\n');

    // Check if canvas is available
    let Canvas;
    try {
        Canvas = require('canvas');
    } catch (e) {
        console.log('❌ Canvas module not found. Installing...\n');
        const { execSync } = require('child_process');
        try {
            execSync('npm install canvas', { stdio: 'inherit' });
            Canvas = require('canvas');
        } catch (err) {
            console.log('\n❌ Failed to install canvas. Creating basic placeholder files instead.\n');
            createPlaceholders();
            return;
        }
    }

    const { createCanvas } = Canvas;

    // Create assets directory if it doesn't exist
    const assetsDir = path.join(__dirname, 'assets');
    if (!fs.existsSync(assetsDir)) {
        fs.mkdirSync(assetsDir, { recursive: true });
    }

    // ADMIN splash: Black background with "PM ADMIN" white text
    console.log('📱 Creating ADMIN splash logo (Black with "PM ADMIN")...');
    createTextLogo(
        createCanvas,
        ['PM', 'ADMIN'],
        '#000000',
        '#FFFFFF',
        path.join(assetsDir, 'splash_admin.png')
    );

    // PARTNER splash: Blue background with "PM Partner" white text
    console.log('📱 Creating PARTNER splash logo (Blue with "PM Partner")...');
    createTextLogo(
        createCanvas,
        ['PM', 'Partner'],
        '#2563EB',
        '#FFFFFF',
        path.join(assetsDir, 'splash_partner.png')
    );

    console.log('\n🎉 Splash logos created successfully!\n');
    console.log('Next steps:');
    console.log('  1. Run: ./generate_admin_assets.sh');
    console.log('  2. Run: ./generate_partner_assets.sh');
    console.log('  3. Test: ./run_admin_app.sh');
    console.log('  4. Test: ./run_partner_app.sh\n');
}

function createTextLogo(createCanvas, textLines, bgColor, textColor, outputPath) {
    const canvas = createCanvas(1024, 1024);
    const ctx = canvas.getContext('2d');

    // Fill background
    ctx.fillStyle = bgColor;
    ctx.fillRect(0, 0, 1024, 1024);

    // Setup text
    ctx.fillStyle = textColor;
    ctx.textAlign = 'center';
    ctx.textBaseline = 'middle';

    // First line (larger)
    ctx.font = 'bold 180px Arial';
    ctx.fillText(textLines[0], 512, 400);

    // Second line (smaller)
    ctx.font = 'bold 150px Arial';
    ctx.fillText(textLines[1], 512, 600);

    // Save to file
    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync(outputPath, buffer);
    console.log(`✅ Created: ${outputPath}`);
}

function createPlaceholders() {
    console.log('Creating simple placeholder files...\n');
    
    const assetsDir = path.join(__dirname, 'assets');
    if (!fs.existsSync(assetsDir)) {
        fs.mkdirSync(assetsDir, { recursive: true });
    }

    // Create minimal 1x1 PNG files as placeholders
    // These are just temporary - user can replace with proper logos later
    
    // Minimal black PNG (1x1)
    const blackPNG = Buffer.from([
        0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A, 0x00, 0x00, 0x00, 0x0D,
        0x49, 0x48, 0x44, 0x52, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01,
        0x08, 0x02, 0x00, 0x00, 0x00, 0x90, 0x77, 0x53, 0xDE, 0x00, 0x00, 0x00,
        0x0C, 0x49, 0x44, 0x41, 0x54, 0x08, 0xD7, 0x63, 0x60, 0x60, 0x60, 0x00,
        0x00, 0x00, 0x04, 0x00, 0x01, 0x27, 0x6B, 0xEE, 0xC7, 0x00, 0x00, 0x00,
        0x00, 0x49, 0x45, 0x4E, 0x44, 0xAE, 0x42, 0x60, 0x82
    ]);

    // Minimal blue PNG (1x1)
    const bluePNG = Buffer.from([
        0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A, 0x00, 0x00, 0x00, 0x0D,
        0x49, 0x48, 0x44, 0x52, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01,
        0x08, 0x02, 0x00, 0x00, 0x00, 0x90, 0x77, 0x53, 0xDE, 0x00, 0x00, 0x00,
        0x0C, 0x49, 0x44, 0x41, 0x54, 0x08, 0xD7, 0x63, 0x64, 0xC8, 0xF0, 0x00,
        0x00, 0x02, 0x85, 0x01, 0x04, 0x8F, 0x6C, 0x8B, 0xC9, 0x00, 0x00, 0x00,
        0x00, 0x49, 0x45, 0x4E, 0x44, 0xAE, 0x42, 0x60, 0x82
    ]);

    fs.writeFileSync(path.join(assetsDir, 'splash_admin.png'), blackPNG);
    console.log('✅ Created: assets/splash_admin.png (placeholder)');

    fs.writeFileSync(path.join(assetsDir, 'splash_partner.png'), bluePNG);
    console.log('✅ Created: assets/splash_partner.png (placeholder)');

    console.log('\n⚠️  Note: These are minimal placeholder files.');
    console.log('   The splash screens will show solid colors without text.');
    console.log('   Install canvas for proper text logos: npm install canvas\n');
}

// Run
generateLogos().catch(err => {
    console.error('Error:', err);
    createPlaceholders();
});

File: apps/playmaker/generate_splash_logos.py
#!/usr/bin/env python3
"""
Generate text-based splash screen logos for Playmaker apps
No external dependencies needed - uses PIL/Pillow which is commonly available
"""

try:
    from PIL import Image, ImageDraw, ImageFont
    import os
except ImportError:
    print("❌ Error: Pillow (PIL) is required. Install with: pip3 install Pillow")
    exit(1)


def create_text_logo(output_path, bg_color, text_lines, text_color=(255, 255, 255)):
    """
    Create a simple text-based logo
    
    Args:
        output_path: Where to save the PNG
        bg_color: RGB tuple for background (e.g., (0, 0, 0) for black)
        text_lines: List of text lines to display
        text_color: RGB tuple for text color (default white)
    """
    # Create 1024x1024 image
    size = 1024
    img = Image.new('RGB', (size, size), bg_color)
    draw = ImageDraw.Draw(img)
    
    # Try to use a bold system font, fall back to default
    try:
        # Try to find Arial Black or similar bold font
        font_large = ImageFont.truetype("/System/Library/Fonts/Supplemental/Arial Bold.ttf", 150)
        font_small = ImageFont.truetype("/System/Library/Fonts/Supplemental/Arial Bold.ttf", 120)
    except:
        try:
            # Try Helvetica Bold
            font_large = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", 150)
            font_small = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", 120)
        except:
            # Fall back to default font
            font_large = ImageFont.load_default()
            font_small = ImageFont.load_default()
    
    # Calculate vertical positioning
    total_height = 0
    line_fonts = []
    for i, line in enumerate(text_lines):
        font = font_large if i == 0 else font_small
        line_fonts.append(font)
        bbox = draw.textbbox((0, 0), line, font=font)
        total_height += (bbox[3] - bbox[1]) + 20  # Add spacing between lines
    
    # Start position (centered vertically)
    y = (size - total_height) // 2
    
    # Draw each line
    for line, font in zip(text_lines, line_fonts):
        bbox = draw.textbbox((0, 0), line, font=font)
        text_width = bbox[2] - bbox[0]
        text_height = bbox[3] - bbox[1]
        x = (size - text_width) // 2
        
        draw.text((x, y), line, fill=text_color, font=font)
        y += text_height + 20
    
    # Save the image
    img.save(output_path, 'PNG')
    print(f"✅ Created: {output_path}")


def main():
    # Create assets directory if it doesn't exist
    assets_dir = "assets"
    if not os.path.exists(assets_dir):
        os.makedirs(assets_dir)
    
    print("🎨 Generating splash screen logos...")
    print("")
    
    # ADMIN splash: Black background with "PM ADMIN" white text
    print("📱 Creating ADMIN splash logo...")
    create_text_logo(
        output_path=os.path.join(assets_dir, "splash_admin.png"),
        bg_color=(0, 0, 0),  # Black
        text_lines=["PM", "ADMIN"],
        text_color=(255, 255, 255)  # White
    )
    
    # PARTNER splash: Blue background with "PM Partner" white text
    print("📱 Creating PARTNER splash logo...")
    create_text_logo(
        output_path=os.path.join(assets_dir, "splash_partner.png"),
        bg_color=(37, 99, 235),  # Blue #2563EB
        text_lines=["PM", "Partner"],
        text_color=(255, 255, 255)  # White
    )
    
    print("")
    print("🎉 Splash logos created successfully!")
    print("")
    print("Next steps:")
    print("  1. Run: ./generate_admin_assets.sh")
    print("  2. Run: ./generate_partner_assets.sh")
    print("  3. Test: ./run_admin_app.sh")
    print("  4. Test: ./run_partner_app.sh")


if __name__ == "__main__":
    main()


